Framework/LiveData/test/KafkaHistoStreamDecoderTest.h

```diff
@@ -0,0 +1,189 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_LIVEDATA_KAFKAHISTOSTREAMDECODERTEST_H_
+#define MANTID_LIVEDATA_KAFKAHISTOSTREAMDECODERTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "KafkaTesting.h"
+#include "MantidAPI/Run.h"
+#include "MantidGeometry/Instrument.h"
+#include "MantidHistogramData/FixedLengthVector.h"
+#include "MantidHistogramData/HistogramX.h"
+#include "MantidHistogramData/HistogramY.h"
+#include "MantidKernel/ConfigService.h"
+#include "MantidKernel/TimeSeriesProperty.h"
+#include "MantidKernel/make_unique.h"
+#include "MantidLiveData/Kafka/KafkaHistoStreamDecoder.h"
+#include <Poco/Path.h>
+
+using Mantid::LiveData::KafkaHistoStreamDecoder;
+using namespace KafkaTesting;
+
+class KafkaHistoStreamDecoderTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static KafkaHistoStreamDecoderTest *createSuite() {
+    return new KafkaHistoStreamDecoderTest();
+  }
+  static void destroySuite(KafkaHistoStreamDecoderTest *suite) { delete suite; }
+
+  void setUp() override {
+    // Temporarily change the instrument directory to the testing one
+    using Mantid::Kernel::ConfigService;
+    auto &config = ConfigService::Instance();
+    auto baseInstDir = config.getInstrumentDirectory();
+    Poco::Path testFile =
+        Poco::Path(baseInstDir)
+            .resolve("IDFs_for_UNIT_TESTING/UnitTestFacilities.xml");
+    // Load the test facilities file
+    config.updateFacilities(testFile.toString());
+    config.setFacility("TEST");
+    // Update instrument search directory
+    config.setString("instrumentDefinition.directory",
+                     baseInstDir + "/IDFs_for_UNIT_TESTING");
+  }
+
+  void tearDown() override {
+    using Mantid::Kernel::ConfigService;
+    auto &config = ConfigService::Instance();
+    config.reset();
+    // Restore the main facilities file
+    config.updateFacilities();
+  }
+
+  void test_Histo_Stream() {
+    using namespace ::testing;
+    using namespace KafkaTesting;
+    using Mantid::API::Workspace_sptr;
+    using Mantid::DataObjects::Workspace2D;
+    using namespace Mantid::LiveData;
+
+    auto mockBroker = std::make_shared<MockKafkaBroker>();
+    EXPECT_CALL(*mockBroker, subscribe_(_, _))
```

Comment:
Not clear to my why this is being called 3 times or why different return types are provided on each call - why is this necessary for the test?